### PR TITLE
make logger names more consistent

### PR DESCRIPTION
**Pull request by dhellmann**
Use consistent capitalization and naming conventions for the loggers.

### CODE REVIEW

File: controllers/metal3.io/baremetalhost_controller_test.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ func newTestReconcilerWithProvisionerFactory(factory provisioner.Factory, initOb
 		Client:             c,
 		Scheme:             scheme.Scheme,
 		ProvisionerFactory: factory,
-		Log:                ctrl.Log.WithName("controller").WithName("baremetalhost").WithName("test"),
+		Log:                ctrl.Log.WithName("controller").WithName("BareMetalHost").WithName("test"),
```
Comment (zaneb) [inline controllers/metal3.io/baremetalhost_controller_test.go:102]: In main.go it's `controllers` with an s.

Alternatively, on line 752 and in host_config_data_test.go it's 'Test' and then the test name.

No idea what's right, but there is some inconsistency remaining.

Comment (dhellmann) [reply controllers/metal3.io/baremetalhost_controller_test.go]: OK, I've moved the test log messages to the logger instantiated as part of the `testing.T` instance and I think I'm using actually consistent names for the "production" code loggers.

**Pull request merged**